### ISSUE TITLE
(bug report summary)

### ISSUE THREAD

**Issue by danieldaeschle** [intellij-v/intellij-v#509]
### Describe the bug

![image](https://user-images.githubusercontent.com/30751516/224546497-ecd88b13-602b-4ad2-8982-43ccb6432b10.png)

### Expected Behavior

Should give an error since string is not assignable to Algorithm (it's an enum)

### Current Behavior

.

### Reproduction Steps

.

### Environment details (`v doctor` output)

.

### IDE Type (IDEA Community/Ultimate, GoLand, CLion, etc.)

CLion

**Comment by danieldaeschle** [comment]
There is also a bug, that you can assign string to enum fields and vice versa.

**Comment by i582** [comment]
his is more likely not a bug, since type checking is generally not implemented, so it is logical that this particular case does not give errors either.

**Comment by danieldaeschle** [comment]
Ah, but it's planned?

**Comment by i582** [comment]
> Ah, but it's planned?

Of course, but later, since correct type inference in all places is not a trivial process
